sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FoldablePropagationSuite.scala

```diff
@@ -118,14 +118,30 @@ class FoldablePropagationSuite extends PlanTest {
       Seq(
         testRelation.select(Literal(1).as('x), 'a).select('x + 'a),
         testRelation.select(Literal(2).as('x), 'a).select('x + 'a)))
-      .select('x)
     val optimized = Optimize.execute(query.analyze)
     val correctAnswer = Union(
       Seq(
         testRelation.select(Literal(1).as('x), 'a).select((Literal(1).as('x) + 'a).as("(x + a)")),
         testRelation.select(Literal(2).as('x), 'a).select((Literal(2).as('x) + 'a).as("(x + a)"))))
-      .select('x).analyze
```

Comment:
We are not checking whether it is resolvable or not. See the PR https://github.com/apache/spark/pull/15417